src/System.Data.SqlClient/src/System/Data/SqlClient/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,979 @@
+            // not allow for a zero byte buffer, we'll create a one byte buffer which, in normal
+            // usage, will cause System.IO.FileStream to utilize the user-supplied buffer and
+            // automatically flush the data directly to the disk cache. In pathological scenarios 
+            // where the user is writing a single byte at a time, we'll explicitly call flush ourselves.
+            if (count == 1)
+            {
+                // calling flush here will mimic the internal control flow of System.IO.FileStream
+                _m_fs.Flush();
+            }
+        }
+
+        public override void WriteByte(byte value)
+        {
+            if (_m_disposed)
+                throw ADP.ObjectDisposed(this);
+
+            _m_fs.WriteByte(value);
+
+            // SQLBUVSTS# 193123 - disable lazy flushing of written data in order to prevent
+            // potential exceptions during Close/Finalization. Since our internal buffer is
+            // only a single byte in length, the provided user data will always be cached.
+            // As a result, we need to be sure to flush the data to disk ourselves.
+
+            // calling flush here will mimic the internal control flow of System.IO.FileStream
+            _m_fs.Flush();
+        }
+
+        #endregion
+
+        static private readonly char[] s_invalidPathChars = Path.GetInvalidPathChars();
+
+        // path length limitations:
+        // 1. path length storage (in bytes) in UNICODE_STRING is limited to UInt16.MaxValue bytes = Int16.MaxValue chars
+        // 2. GetFullPathName API of kernel32 does not accept paths with length (in chars) greater than 32766
+        //  (32766 is actually Int16.MaxValue - 1, while (-1) is for NULL termination)
+        // We must check for the lowest value between the the two
+        private const int MaxWin32PathLength = short.MaxValue - 1;
+
+        [Conditional("DEBUG")]
+        static private void AssertPathFormat(string path)
+        {
+            Debug.Assert(path != null);
+            Debug.Assert(path == path.Trim());
+            Debug.Assert(path.Length > 0);
+            Debug.Assert(path.Length <= MaxWin32PathLength);
+            Debug.Assert(path.IndexOfAny(s_invalidPathChars) < 0);
+            Debug.Assert(path.StartsWith(@"\\", StringComparison.OrdinalIgnoreCase));
+            Debug.Assert(!path.StartsWith(@"\\.\", StringComparison.Ordinal));
+        }
+
+        // SQLBUVSTS01 bugs 192677 and 193221: we cannot use System.IO.Path.GetFullPath for two reasons:
+        // * it requires PathDiscovery permissions, which is unnecessary for SqlFileStream since we are dealing with network path
+        // * it is limited to 260 length while in our case file path can be much longer
+        // To overcome the above limitations we decided to use GetFullPathName function from kernel32.dll
+        [ResourceExposure(ResourceScope.Machine)]
```

Comment:
I don't think these are useful- @stephentoub ?

Reply:
@ do we need these?

---
In reply to: [190732837](https://github.com/dotnet/corefx/pull/29879#discussion_r190732837) [](ancestors = 190732837)